### PR TITLE
Integrate threat logic with backend

### PR DESCRIPTION
**Pull request by RichardLam772001**
https://www.loom.com/share/8d08d2ae6a3140de84a6121814b88a61

Also ticket made to have threat not spawn in user room:
https://github.com/RichardLam772001/cs375/issues/77

Didn't wanna make this PR too big. Accomplishes displaying the threat on the frontend.

Making a threat disappear after a user addresses it will be in a follow up ticket (probably the player actions stuff)

### CODE REVIEW

File: app/game/game.js [hunk 1]
```diff
@@ -28,6 +28,8 @@ const GAME = (humanUsername, aiUsername, gameId) => {
     const THREATS = [];
     const AVAILABLE_ROOMS = ["0-0", "0-1", "0-2", "1-0", "1-1", "1-2", "2-0", "2-1", "2-2"]; // Richard: Yes I know it's hardcoded, we can make a dynamic room generator later TO DO 
     const ROOMS_WITH_THREATS = [];
+    const THREAT_TYPES = ["fire", "breach", "invader"];
+    const MAX_THREATS_PER_ROOM = 3;
```
Comment (ColinJPage) [inline app/game/game.js:32]: I think you mean `MAX_ACTIVE_THREATS`, because the max threats per room is 1

File: app/game/game.js [hunk 2]
```diff
@@ -72,22 +77,50 @@ const GAME = (humanUsername, aiUsername, gameId) => {
 
     const spawnThreat = () => {
         const threatRoom = selectThreatRoom(AVAILABLE_ROOMS, ROOMS_WITH_THREATS);
-        const threat = Threat(() => onThreatUnresolved(threatRoom, THREATS.length));
-        THREATS.push(threat);
+        const threat = Threat(randomlySelectThreat(), () => onThreatUnresolved(threatRoom, THREATS.length));
```
Comment (ColinJPage) [inline app/game/game.js:80]: When the threat is unresolved, it removes the threat at the index `THREATS.length`. The assumption is that the index of this threat won't change between now and when it is removed. But this is not true. An earlier threat in the array may be removed, meaning the index has changed and the index now corresponds to the wrong threat.

File: app/game/game.js [hunk 3]
```diff
@@ -72,22 +77,50 @@ const GAME = (humanUsername, aiUsername, gameId) => {
...
+    }
+    /**
+     * @param {string} room e.g. 0-0 to index with 
+     */
+    const alertHumanPlayerOfThreat = (room) => {
+        alertPlayerOfThreat(THREATS[room], HUMAN_USERNAME, room);
     }
 
     const onThreatUnresolved = (room, threatId) => {
         THREATS.splice(threatId, 1); // Remove threat from THREATS list
```
Comment (ColinJPage) [inline app/game/game.js:112]: Double-check this

File: app/public/human/human.js [hunk 4]
```diff
@@ -1,34 +1,54 @@
...
-                }
-            },
-            username: USERNAME_COOKIE,
-            gameId: GAME_ID_COOKIE
-        });
-    };
+  const useTool = (tool) => {
+    WS.send({
+      action: {
+        name: "useTool",
```
Comment (ColinJPage) [inline app/public/human/human.js:18]: I recommend changing this to "switchTool". Remember, using a tool is automatic when entering a threatened room with the right tool

**Review by ColinJPage — CHANGES_REQUESTED** [review]
Looks good, but there's some stuff I want you to double-check.
Also, I'd like us to represent rooms as objects instead of strings, because we wouldn't have to parse them every time. I make that change in #64 , so I can resolve those conflicts once you merge this.

File: app/game/game.js (hunk 2 above)
Comment (RichardLam772001) [reply app/game/game.js:80]: Now threat is indexed by room

**Review by ColinJPage — APPROVED** [review]
Nice!

**Pull request merged**